### DISCUSSION TITLE
[feature request] include escape sequence in `write_screen_file`

### DISCUSSION THREAD

**Discussion by xzbdmw** [ghostty-org/ghostty#3841] [Feature Requests, Ideas] [answered]
So I can open it in nvim and use `:term cat **.log` to have colors(currently no color) plus vim motion(this is already achievable).

**Comment by 0x4a61636f62** [comment]
This discussion might be related:

https://github.com/ghostty-org/ghostty/discussions/3708

**Reply by xzbdmw** [reply]
Yeah if it can includes colors, open in nvim with all my plugins is perfect than, say,  ghostty implements its own vim mode.

**Comment by 00-kat** [comment]
It will look horrible if the editor doesn't do anything special with those escape sequences.

Example in Kitty, which *does* preserve escape sequences:
![image](https://github.com/user-attachments/assets/9aca3883-a018-4354-a716-4abbb8685658)

**Reply by xzbdmw** [reply]
You can open the builtin terminal and run `cat filename`
or use `nvim_open_term` then `nvim_chan_send` to display color, along with full vim motion.

What I mean is to add a `include escape sequence` version of scorllback, not replacing current pure text one.

**Reply by 00-kat** [reply]
> You can open the builtin terminal and run `cat filename`

Sure, but my point was that not every editor does special things with the escape sequences.  For example, Kakoune and Helix don't have a terminal emulator.

> [rest]

Got it, cool.

**Comment by 00-kat** [comment]
#5555 is a duplicate of this but based on the resolution of #5555:

> I think this can be rolled in as a comment to the issue discussion for retaining control codes for copy.

I believe this should be closed.

**Comment by kaizoplant** [comment]
okay sorry for asking late but where is the said issue discussion? ive been trying to find it for a while and i dont know exactly where it is,

**Reply by xzbdmw** [reply]
me too. This relationship of discussion and issue is weird.

**Reply by 00-kat** [reply]
Hm.  I thought I remembered finding it but I can't find it now either.  Perhaps this should be reopened until someone finds it.

**Comment by pickx** [comment]
can we open an issue for this? this seems pretty useful.

I'd like to try and implement this (though I have no idea how difficult it would be)

**Comment by mitchellh** [comment, marked as answer]
#9398 accepted.

**Comment by hayesgm** [comment]
This looks great, except that the full control-sequence set supported by Ghostty isn’t cross-compatible with the more limited ANSI 16-color subset used by certain clients (e.g., Discord). I’m wondering whether it would make sense to support an additional output format, something like `"copy_to_clipboard:ansi16"` or whether that feels too client-specific for Ghostty’s scope. I'd be happy to do the legwork if so.
